samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -1084,7 +1086,7 @@ def _link_gateway_resource_to_gateway_rest_apis_rest_api_id_call_back(
     """
     # if the destination rest api list contains more than one element, so we have an issue in our linking logic
     if len(referenced_rest_apis_values) > 1:
-        raise InvalidResourceLinkingException("Could not link multiple Rest APIs to one Gateway resource")
+        raise InvalidResourceLinkingException("Could not link multiple Rest APIs to one Gateway method resource")
```

Comment:
we use this call back to link different Gateway resources (GatewayResource, GatewayMethod, and I believe different other resources), that is why we tried to make the exception message generic